src/app/shared/search-box/search-box.component.ts

```diff
@@ -181,12 +180,14 @@ export class SearchBoxComponent implements OnInit, AfterViewInit {
       return Observable.of(suggestions);
     }
     suggestions.links = [];
-    for (let link of this.linkTypeService.getLinkTypes()) {
-      if (link.name.toLowerCase().startsWith(this.text.toLowerCase())) {
-        suggestions.links.push(link);
-      }
-    }
-    return Observable.of(suggestions);
+
+    return this.linkTypeService.getLinkTypes({}).pipe(
+      map(linkTypes => {
+        const linkTypeSuggestions = linkTypes.filter(linkType => linkType.name.toLowerCase().startsWith(this.text.toLowerCase()));
+        suggestions.links.push(...linkTypeSuggestions);
+        return suggestions;
```

Comment:
Why not just return `linkTypeSuggestions`?

Reply:
Because we need to return all the suggestions (including the already existing ones in that array). See the removed code above.

Reply:
Ok.
The `suggestions.links = [];` confused me as clearing `suggestions`. (Which just clears `suggestions.links` and leaves the rest of `suggestions` untouched.)